src/common/ownsql.cpp

```diff
@@ -132,28 +141,29 @@ bool SqlDatabase::openOrCreateReadWrite(const QString &filename)
 
     auto checkResult = checkDb();
     if (checkResult != CheckDbResult::Ok) {
+        close();
         if (checkResult == CheckDbResult::CantPrepare) {
             // When disk space is low, preparing may fail even though the db is fine.
             // Typically CANTOPEN or IOERR.
             qint64 freeSpace = Utility::freeDiskSpace(QFileInfo(filename).dir().absolutePath());
             if (freeSpace != -1 && freeSpace < 1000000) {
                 qCWarning(lcSql) << "Can't prepare consistency check and disk space is low:" << freeSpace;
-                close();
-                return false;
-            }
-
-            // Even when there's enough disk space, it might very well be that the
-            // file is on a read-only filesystem and can't be opened because of that.
-            if (_errId == SQLITE_CANTOPEN) {
+            } else if (_errId == SQLITE_CANTOPEN) {
+                // Even when there's enough disk space, it might very well be that the
+                // file is on a read-only filesystem and can't be opened because of that.
                 qCWarning(lcSql) << "Can't open db to prepare consistency check, aborting";
-                close();
-                return false;
+            } else if (_errId == SQLITE_LOCKED || _errId == SQLITE_BUSY) {
+                qCWarning(lcSql) << "Can't open db to prepare consistency check, the db is locked aborting" << _errId << _error;
             }
+            return false;
         }
 
         qCCritical(lcSql) << "Consistency check failed, removing broken db" << filename;
-        close();
-        QFile::remove(filename);
+        QFile fileToRemove(filename);
+        if (!fileToRemove.remove()) {
```

Comment:
If removing the broken db failed we are not ready...

Reply:
No just return false, the code was meant to fix a corrupted db by removing it.
However if removing the broken db fails.... this pr provides the required error handling.